werkzeug/serving.py

```diff
@@ -148,7 +148,11 @@ def write(data):
                     self.send_header('Date', self.date_time_string())
                 self.end_headers()
 
-            assert type(data) is bytes, 'applications must write bytes'
+            assert isinstance(data, bytes),\
+                    'applications must write bytes - found \'%s\'' % type(data)
+# relax the strict `type(data) is bytes` assertion in order to allow subclasses
```

Comment:
the idea of that assertion is that it only fires when you do something thats broken,
as far as i'm concerned extra type information is not necessary at that place